dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/patch/AbstractResourcePatch.java

```diff
@@ -26,85 +22,86 @@
-
-        List<Operation> operations = patch.getOperations();
+    public RestModel patch(R restModel, List<Operation> operations)
+            throws UnprocessableEntityException, PatchBadRequestException {
 
         // Note: the list of possible operations is taken from JsonPatchConverter class. Does not implement
         // test https://tools.ietf.org/html/rfc6902#section-4.6
         ops: for (Operation op : operations) {
             switch (op.getOp()) {
                 case "add":
-                    add(restModel, context, op);
+                    restModel = add(restModel, op);
                     continue ops;
                 case "replace":
-                    replace(restModel, context, op);
+                    restModel = replace(restModel, op);
                     continue ops;
                 case "remove":
-                    remove(restModel, context, op);
+                    restModel = remove(restModel, op);
                     continue ops;
                 case "copy":
-                    copy(restModel, context, op);
+                    restModel = copy(restModel, op);
                     continue ops;
                 case "move":
-                    move(restModel, context, op);
+                    restModel = move(restModel, op);
                     continue ops;
                 default:
                     // JsonPatchConverter should have thrown error before this point.
                     throw new PatchBadRequestException("Missing or illegal patch operation: " + op.getOp());
             }
         }
+
+        return restModel;
+
     }
     // The default patch methods throw an error when no sub-class implementation is provided.
 
-    protected void add(R restModel, Context context, Operation operation)
-        throws UnprocessableEntityException, PatchBadRequestException, SQLException, AuthorizeException {
+    protected R add(R restModel, Operation operation)
+            throws UnprocessableEntityException, PatchBadRequestException {
         throw new UnprocessableEntityException(
-            "The add operation is not supported."
+                "The add operation is not supported."
         );
     }
 
-    protected void replace(R restModel, Context context, Operation operation)
-        throws UnprocessableEntityException, PatchBadRequestException, SQLException, AuthorizeException {
+    protected R replace(R restModel, Operation operation)
+            throws UnprocessableEntityException, PatchBadRequestException {
         // The replace operation is functionally identical to a "remove" operation for
         // a value, followed immediately by an "add" operation at the same
         // location with the replacement value. https://tools.ietf.org/html/rfc6902#section-4.3
-        remove(restModel, context, operation);
-        add(restModel, context, operation);
+       // remove(restModel, context, operation);
+       // add(restModel, context, operation);
```

Comment:
As you commented these out, it looks like we should remove all 5 lines of comments here, as they seem inapplicable?